src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -151,9 +154,6 @@ class BaseSearchInput extends Component<SearchInputProps> {
       if (!('value' in this.props)) {
```

Comment:
I think debounce should not be applied in controlled state because this leads to weird behaviour of not updating the input value immediately. Not sure if we should tackle this or leave it up for the teams using this?

Reply:
I think that's something that can be left to the project teams. Might be good to mention it in the prop description though.

Reply:
True that, probably better not to touch it, but mention in the docs somehow.

Reply:
Or just leave it out, it will be obvious when testing it. One would have to be ignorant of controlled pattern altogether if using with debounce.